### PR TITLE
feat(related-products): add selectors and facade

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
Adds selectors and facade for related product state

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/related-products/state/src/facades/related-product-facade.interface.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import { DaffStoreFacade } from '@daffodil/core/state';
+import { DaffProduct } from '@daffodil/product';
+
+/**
+ * A facade for getting state about a particular product's related products.
+ */
+export interface DaffRelatedProductsFacadeInterface<T extends DaffProduct = DaffProduct> extends DaffStoreFacade<Action> {
+	/**
+	 * A list of related products.
```
Comment (lderrickable) [inline libs/related-products/state/src/facades/related-product-facade.interface.ts:12]: This should be "A list of the related products of the current product."

File: libs/related-products/state/src/selectors/related-products/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  DaffRelatedProductStateRootSlice,
+} from '../../reducers/reducers-state.interface';
+import { getDaffRelatedProductsFeatureSelector } from '../feature.selector';
+
+/**
+ * An interface for selectors related to the current product page.
+ */
+export interface DaffRelatedProductsMemoizedSelectors<T extends DaffProduct = DaffProduct> {
+	/**
+	 * Selects the id of the current product.
```
Comment (lderrickable) [inline libs/related-products/state/src/selectors/related-products/selectors.ts:21]: wrong comment

File: libs/related-products/state/src/selectors/related-products/selectors.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  );
+
+  return {
+    selectRelatedProductIds,
+    selectRelatedProducts,
+  };
+};
+
+/**
+ * A function that returns all selectors related to the product page.
```
Comment (lderrickable) [inline libs/related-products/state/src/selectors/related-products/selectors.ts:62]: wrong comment

File: libs/related-products/state/testing/src/mock-related-products.facade.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import { BehaviorSubject } from 'rxjs';
+
+import { DaffProduct } from '@daffodil/product';
+import { DaffRelatedProductsFacadeInterface } from '@daffodil/related-products/state';
+
+/**
+ * @inheritdoc
+ */
+@Injectable({ providedIn: 'root' })
+export class MockDaffRelatedProductsFacade implements DaffRelatedProductsFacadeInterface {
```
Comment (lderrickable) [inline libs/related-products/state/testing/src/mock-related-products.facade.ts:11]: I think this needs an extra comment to explain what the mock facade is for.

File: libs/related-products/state/src/facades/related-product-facade.interface.ts (hunk 1 above)
Comment (griest024) [reply libs/related-products/state/src/facades/related-product-facade.interface.ts]: changed

File: libs/related-products/state/src/selectors/related-products/selectors.ts (hunk 2 above)
Comment (griest024) [reply libs/related-products/state/src/selectors/related-products/selectors.ts]: changed

File: libs/related-products/state/src/selectors/related-products/selectors.ts (hunk 3 above)
Comment (griest024) [reply libs/related-products/state/src/selectors/related-products/selectors.ts]: changed

File: libs/related-products/state/testing/src/mock-related-products.facade.ts (hunk 4 above)
Comment (griest024) [reply libs/related-products/state/testing/src/mock-related-products.facade.ts:13]: changed

**Review by lderrickable — APPROVED** [review]

**Pull request merged**